### DISCUSSION TITLE
셀프 리뷰를 위한 코드 컨벤션 정리

### DISCUSSION THREAD

**Discussion by kmularise** [dailyge/dailyge-server#193] [Decision]
## 📝 연관 PR

<br/><br/>

## ❓ 작성 이유

코드 스타일의 차이로 인해 코드 리뷰가 복잡해졌습니다. 또한 셀프 코드 리뷰를 할 때에도 빠뜨리는 항목이 발생하는 경우가 있었습니다. 이에 대해 코드 컨벤션을 문서화해 코드 리뷰 시간을 단축하고 꼼꼼한 셀프리뷰를 진행할 수 있게끔 합니다.

<br/><br/>

## 📝 내용
* 원격저장소에 올리기 전에는 수정한 모듈(dailyge-api, admin-api)의 pmd, checkstyle을 진행합니다.
* 메소드 관련해서 파라미터가 1개일때는 개행을 안해도 되지만, 파라미터가 2개이상이면 개행을 해야 합니다.
* 변하지 않는 변수 앞에는 final을 붙여야 합니다. 다만 인터페이스 메소드 파라미터에서는 final을 붙이지 않아도 됩니다.
* 어노테이션을 붙이는 변수의 경우에는 어노테이션, final 순서대로 씁니다.
* 어노테이션이 붙은 클래스 변수의 경우 앞뒤로 개행 두번해 띄워줍니다.
* 어노테이션이 붙지 않은 클래스 변수는 개행을 한번만 합니다.
* @RequestMapping의 API의 uri에는 path을 붙여줍니다.
* API 클래스의 request, response 객체는 request, response라 명명한다.
* DTO 객체에는 로그 확인을 위해 toString()를 재정의해줍니다.
* 앞의 space 네칸을 유지합니다.
* service, DAO는 default class로 정의합니다. 추상화를 통해 느슨한 결합을 유지하기 위해서입니다.
* 어노테이션이 두개 이상이면 긴 어노테이션을 아래로 합니다.
* 통합테스트 시에는 class Displayname을 `[IntegrationTest]`으로 시작합니다.
* code and message enum 순서는 상태 코드 순서대로 작성합니다.

```java
    DUPLICATED_WINNER_SELECTION(409, "이미 실행된 당첨자 선정 작업입니다."),
    COUPON_UN_RESOLVED_EXCEPTION(500, "쿠폰 관련 작업이 실패했습니다.");
```

* 인터페이스 메소드 선언 후 개행을 두번 합니다.
<br/><br/>

### 10.07 추가
* 엔티티의 경우, 기본 생성자 접근 권한을 protected 로 합니다. 프록시 조회로 접근 권한을 protected이거나 그보다 더 넓은 범위로 해야 합니다. 접근 권한을 최소화하기 위해 protected로 둡니다.

**Comment by beatmeJY** [comment]
코드 컨벤션! 팀원들을 위해 정리까지 해주셨군요 감사합니다!🚀

**Comment by devjun10** [comment]
이거 댓글을 깜빡했네요. 감사합니다 ㅎㅎ 이런거 시간내서 적기 꽤 힘든데, 덕분에 깔끔하게 정리 됐네요. 더 좋은거 나오면 계속 추가해봐요~ ㅅㅅㅅㅅㅍ노바
